[AG-13140] Fix(options): aggregations column filters allow fuzzy search validation and console warnings for missing custom agg funcs

sdwvit opened the pull request:
Fix aggregation function validation and improve warning messages
 - Ensure only valid aggregation functions are added to the map.
 - Replace error logging with a warning for unrecognized aggregation functions.
 - Enhance warning message to suggest possible valid functions.
 - use vanilla algorithm for string distance

Fixes [AG-13140](https://ag-grid.atlassian.net/browse/AG-13140)
Fixes [AG-15499](https://ag-grid.atlassian.net/browse/AG-15499)

plnkr to check https://plnkr.co/edit/3533WfBrZsng5dDv?open=main.ts&preview (local)

<img width="1104" height="286" alt="image" src="https://github.com/user-attachments/assets/0f3fc5d1-3326-4be6-983a-e4d1213f558b" />

sdwvit committed:
- use vanilla algorithm for string distance

sdwvit committed:
- use vanilla algorithm for string distance

sdwvit committed:
Merge remote-tracking branch 'origin/sdwvit/fix/options/aggregations-column-filters-allow-fuzzy-search-validation-and-console-warnings-for-missing-custom-agg-funcs/AG-13140' into sdwvit/fix/options/aggregations-column-filters-allow-fuzzy-search-validation-and-console-warnings-for-missing-custom-agg-funcs/AG-13140

# Conflicts:
#	packages/ag-grid-community/src/utils/fuzzyMatch.ts

sdwvit committed:
- rename functino

sdwvit committed:
add memory pressure test

sdwvit committed:
lint

sdwvit committed:
lint

sdwvit committed:
Merge branch 'latest' into sdwvit/fix/options/aggregations-column-filters-allow-fuzzy-search-validation-and-console-warnings-for-missing-custom-agg-funcs/AG-13140

sdwvit commented:
AG-13140

sdwvit committed:
Fix aggregation function validation and improve warning messages

 - Ensure only valid aggregation functions are added to the map.
 - Replace error logging with a warning for unrecognized aggregation functions.
 - Enhance warning message to suggest possible valid functions.

gportela85 reviewed (changes requested):
The Levenshtein Distance alone will not be sufficient for this task. We need a combination of weighted match + Levenshtein Distance.

There are also broken tests here.

sdwvit committed:
Enhance fuzzy matching logic and tests

 - Introduce secondary scoring for case-insensitive and consecutive matches
 - Favor matches at the start of the string in scoring
 - Adjust performance test to handle smaller string sizes

sdwvit committed:
update test

sdwvit committed:
add a regression test for exact match filtering out

sdwvit committed:
bump validation module size
